### PR TITLE
Added origin to get_hexagonal_prism(...) function

### PR DESCRIPTION
**Pull request by samuelshaner**
This PR adds an origin attribute to the `get_hexagonal_prism(...)` function to bring it inline with the functionality of the `get_rectangular_prism(...)` function. The origin attribute is added to the list of attributes in the `get_hexagonal_prism(...)` function in the equivalent attribute location as the origin attribute in the list of attributes in the `get_rectangular_prism(...)` function.

### CODE REVIEW

File: openmc/surface.py [hunk 1]
```diff
@@ -2090,23 +2092,26 @@ def get_hexagonal_prism(edge_length=1., orientation='y',
     """
 
     l = edge_length
+    x = origin[0]
+    y = origin[1]
```
Comment (paulromano) [inline openmc/surface.py]: `x, y = origin`

File: openmc/surface.py [hunk 2]
```diff
@@ -2090,23 +2092,26 @@ def get_hexagonal_prism(edge_length=1., orientation='y',
...
 
     if orientation == 'y':
-        right = XPlane(x0=sqrt(3.)/2*l, boundary_type=boundary_type)
-        left = XPlane(x0=-sqrt(3.)/2*l, boundary_type=boundary_type)
+        right = XPlane(x0=x + sqrt(3.)/2*l, boundary_type=boundary_type)
+        left  = XPlane(x0=x - sqrt(3.)/2*l, boundary_type=boundary_type)
```
Comment (paulromano) [inline openmc/surface.py]: PEP8: Always use single space around assignment operator (=), i.e. should be `left = XPlane...`

File: openmc/surface.py [hunk 3]
```diff
@@ -2116,21 +2121,23 @@ def get_hexagonal_prism(edge_length=1., orientation='y',
             lower_right.periodic_surface = upper_left
 
     elif orientation == 'x':
-        top = YPlane(y0=sqrt(3.)/2*l, boundary_type=boundary_type)
-        bottom = YPlane(y0=-sqrt(3.)/2*l, boundary_type=boundary_type)
+        top    = YPlane(y0=y + sqrt(3.)/2*l, boundary_type=boundary_type)
```
Comment (paulromano) [inline openmc/surface.py]: `top = YPlane...`

File: openmc/surface.py [hunk 4]
```diff
@@ -2154,19 +2161,19 @@ def get_hexagonal_prism(edge_length=1., orientation='y',
...
-            x_max_y_min_out = cyl2(name='x max y min out', x0=-l/2, y0=c*l)
-            x_max_y_max_out = cyl2(name='x max y max out', x0=l/2, y0=c*l)
-            x_min_out = cyl2(name='x min out', x0=-l)
-            x_max_out = cyl2(name='x max out', x0=l)
+            x_min_y_min_in = cyl1(name='x min y min in', x0=x-t/2, y0=y-c*t)
+            x_min_y_max_in = cyl1(name='x min y max in', x0=x+t/2, y0=y-c*t)
+            x_max_y_min_in = cyl1(name='x max y min in', x0=x-t/2, y0=y+c*t)
+            x_max_y_max_in = cyl1(name='x max y max in', x0=x+t/2, y0=y+c*t)
+            x_min_in       = cyl1(name='x min in'      , x0=x-t  , y0=y)
+            x_max_in       = cyl1(name='x max in'      , x0=x+t  , y0=y)
```
Comment (paulromano) [inline openmc/surface.py]: PEP8 forbids extra spaces as they are being used here:
```Python
x_min_in = cyl1(name='x min in', x0=x-t, y0=y)
x_max_in = cyl1(name='x max in', x0=x+t, y0=y)
```

File: openmc/surface.py [hunk 5]
```diff
@@ -2154,19 +2161,19 @@ def get_hexagonal_prism(edge_length=1., orientation='y',
...
+
+            x_min_y_min_out = cyl2(name='x min y min out', x0=x-l/2, y0=y-c*l)
+            x_min_y_max_out = cyl2(name='x min y max out', x0=x+l/2, y0=y-c*l)
+            x_max_y_min_out = cyl2(name='x max y min out', x0=x-l/2, y0=y+c*l)
+            x_max_y_max_out = cyl2(name='x max y max out', x0=x+l/2, y0=y+c*l)
+            x_min_out       = cyl2(name='x min out'      , x0=x-l  , y0=y)
+            x_max_out       = cyl2(name='x max out'      , x0=x+l  , y0=y)
```
Comment (paulromano) [inline openmc/surface.py]: Same comment here: don't use extra spaces

File: openmc/surface.py [hunk 6]
```diff
@@ -2176,19 +2183,19 @@ def get_hexagonal_prism(edge_length=1., orientation='y',
...
-            x_max_y_min_out = cyl2(name='x max y min out', x0=c*l, y0=-l/2)
-            x_max_y_max_out = cyl2(name='x max y max out', x0=c*l, y0=l/2)
-            y_min_out = cyl2(name='y min out', y0=-l)
-            y_max_out = cyl2(name='y max out', y0=l)
+            x_min_y_min_in = cyl1(name='x min y min in', x0=x-c*t, y0=y-t/2)
+            x_min_y_max_in = cyl1(name='x min y max in', x0=x-c*t, y0=y+t/2)
+            x_max_y_min_in = cyl1(name='x max y min in', x0=x+c*t, y0=y-t/2)
+            x_max_y_max_in = cyl1(name='x max y max in', x0=x+c*t, y0=y+t/2)
+            y_min_in       = cyl1(name='y min in'      , x0=x    , y0=y-t)
+            y_max_in       = cyl1(name='y max in'      , x0=x    , y0=y+t)
```
Comment (paulromano) [inline openmc/surface.py]: Same comment here: don't use extra spaces

File: openmc/surface.py [hunk 7]
```diff
@@ -2176,19 +2183,19 @@ def get_hexagonal_prism(edge_length=1., orientation='y',
...
+
+            x_min_y_min_out = cyl2(name='x min y min out', x0=x-c*l, y0=y-l/2)
+            x_min_y_max_out = cyl2(name='x min y max out', x0=x-c*l, y0=y+l/2)
+            x_max_y_min_out = cyl2(name='x max y min out', x0=x+c*l, y0=y-l/2)
+            x_max_y_max_out = cyl2(name='x max y max out', x0=x+c*l, y0=y+l/2)
+            y_min_out       = cyl2(name='y min out'      , x0=x      , y0=y-l)
+            y_max_out       = cyl2(name='y max out'      , x0=x      , y0=y+l)
```
Comment (paulromano) [inline openmc/surface.py]: Same comment here: don't use extra spaces

**Comment by samuelshaner** [comment]
Thanks for the quick review @paulromano. I just removed the commits related to adding a cylindrical lattice (I have them on my cyl-lattice branch) so the log should now just reflect the changes in this PR.

**Pull request merged**